avm/res/azure-stack-hci/cluster/deployment-settings/main.bicep

```
@@ -0,0 +1,213 @@
+metadata name = 'Azure Stack HCI Cluster Deployment Settings'
+metadata description = 'This module deploys an Azure Stack HCI Cluster Deployment Settings resource.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Required. The name of the Azure Stack HCI cluster - this must be a valid Active Directory computer name and will be the name of your cluster in Azure.')
```

```suggestion
@description('Conditional. The name of the Azure Stack HCI cluster - this must be a valid Active Directory computer name and will be the name of your cluster in Azure. Required if the template is used in a standalone deployment.')
```

For child-modules, we usually define the parent references as 'Conditional' as they're only required if you deploy this as a standalone template. If you deploy it via it's parent (which is the default mode of deployment from the public registry), the parent cluster module will pass its value down automatically.